option.go

```diff
@@ -0,0 +1,244 @@
+	iana.DEC_ALPHA:         "undionly.kpxe",
+	iana.ARC_X86:           "undionly.kpxe",
+	iana.INTEL_LEAN_CLIENT: "undionly.kpxe",
+	iana.EFI_IA32:          "ipxe.efi",
+	iana.EFI_X86_64:        "ipxe.efi",
+	iana.EFI_XSCALE:        "ipxe.efi",
+	iana.EFI_BC:            "ipxe.efi",
+	iana.EFI_ARM32:         "snp.efi",
+	iana.EFI_ARM64:         "snp.efi",
+	iana.EFI_X86_HTTP:      "ipxe.efi",
+	iana.EFI_X86_64_HTTP:   "ipxe.efi",
+	iana.EFI_ARM32_HTTP:    "snp.efi",
+	iana.EFI_ARM64_HTTP:    "snp.efi",
+	iana.Arch(41):          "snp.efi", // arm rpiboot: https://www.iana.org/assignments/dhcpv6-parameters/dhcpv6-parameters.xhtml#processor-architecture
+}
+
+// String function for clientType.
+func (c clientType) String() string {
+	return string(c)
+}
+
+// String function for UserClass.
+func (c UserClass) String() string {
+	return string(c)
+}
+
+// setDHCPOpts takes a client dhcp packet and data (typically from a backend) and creates a slice of DHCP packet modifiers.
+// m is the DHCP request from a client. d is the data to use to create the DHCP packet modifiers.
+// This is most likely the place where we would have any business logic for determining DHCP option setting.
+func (s *Server) setDHCPOpts(_ context.Context, _ *dhcpv4.DHCPv4, d *data.Dhcp) []dhcpv4.Modifier {
+	mods := []dhcpv4.Modifier{
+		dhcpv4.WithLeaseTime(d.LeaseTime),
+		dhcpv4.WithYourIP(d.IPAddress.IPAddr().IP),
+	}
+	if len(d.NameServers) > 0 {
+		mods = append(mods, dhcpv4.WithDNS(d.NameServers...))
+	}
+	if len(d.DomainSearch) > 0 {
+		mods = append(mods, dhcpv4.WithDomainSearchList(d.DomainSearch...))
+	}
+	if len(d.NTPServers) > 0 {
+		mods = append(mods, dhcpv4.WithOption(dhcpv4.OptNTPServers(d.NTPServers...)))
+	}
+	if !d.BroadcastAddress.IsZero() {
+		mods = append(mods, dhcpv4.WithGeneric(dhcpv4.OptionBroadcastAddress, d.BroadcastAddress.IPAddr().IP))
+	}
+	if d.DomainName != "" {
+		mods = append(mods, dhcpv4.WithGeneric(dhcpv4.OptionDomainName, []byte(d.DomainName)))
+	}
+	if d.Hostname != "" {
+		mods = append(mods, dhcpv4.WithGeneric(dhcpv4.OptionHostName, []byte(d.Hostname)))
+	}
+	if len(d.SubnetMask) > 0 {
+		mods = append(mods, dhcpv4.WithNetmask(d.SubnetMask))
+	}
+	if !d.DefaultGateway.IsZero() {
+		mods = append(mods, dhcpv4.WithRouter(d.DefaultGateway.IPAddr().IP))
+	}
+
+	return mods
+}
+
+// setNetworkBootOpts purpose is to sets 3 or 4 values. 2 DHCP headers, option 43 and optionally option (60).
+//
+// DHCP Headers (https://datatracker.ietf.org/doc/html/rfc2131#section-2)
+// 'siaddr': IP address of next bootstrap server. represented below as `.ServerIPAddr`.
+// 'file': Client boot file name. represented below as `.BootFileName`.
+//
+// DHCP option
+// option 60: Class Identifier. https://www.rfc-editor.org/rfc/rfc2132.html#section-9.13
+// option 60 is set if the client's option 60 (Class Identifier) starts with HTTPClient.
+func (s *Server) setNetworkBootOpts(ctx context.Context, m *dhcpv4.DHCPv4, n *data.Netboot) func(d *dhcpv4.DHCPv4) {
```

Comment:
It was not obvious to me why this function encloses and returns another function.
I did figure out that this is to return something that can act as a `github.com/insomniacslk/dhcp/dhcpv4.Modifier`
This does seem to be successfully wrapping the caller's context rather than the `Server`'s context, perhaps this is a sign that the server doesn't need its own context after all?

Reply:
That's a fair point. The context in the `Server` struct will actually be for the handlers to consume. I've actually already built out the handler functionality (see #3 ) and then broke out the code into pieces for PR reviews. I think I carried this `Server.ctx` over and as you've noted its definitely not needed in this PR.